lib/python/Tools/KeyBindings.py

```diff
@@ -1,277 +1,379 @@
+	# but the remote.xml file issues KEY_PREVIOUSSONG
+	# and KEY_NEXTSONG.
+	KEYIDS["BTN_0"]: ("UP", "fp"),
+	KEYIDS["BTN_1"]: ("DOWN", "fp"),
+	KEYIDS["KEY_0"]: ("0",),
+	KEYIDS["KEY_1"]: ("1",),
+	KEYIDS["KEY_2"]: ("2",),
+	KEYIDS["KEY_3"]: ("3",),
+	KEYIDS["KEY_4"]: ("4",),
+	KEYIDS["KEY_5"]: ("5",),
+	KEYIDS["KEY_6"]: ("6",),
+	KEYIDS["KEY_7"]: ("7",),
+	KEYIDS["KEY_8"]: ("8",),
+	KEYIDS["KEY_9"]: ("9",),
+	KEYIDS["KEY_AUDIO"]: ("AUDIO",),
+	KEYIDS["KEY_BACK"]: ("RECALL",),
+	KEYIDS["KEY_BLUE"]: ("BLUE",),
+	KEYIDS["KEY_BOOKMARKS"]: ("PLAYLIST",),
+	KEYIDS["KEY_CHANNELDOWN"]: ("BOUQUET-",),
+	KEYIDS["KEY_CHANNELUP"]: ("BOUQUET+",),
+	KEYIDS["KEY_CONTEXT_MENU"]: ("CONTEXT",),
+	KEYIDS["KEY_DOWN"]: ("DOWN",),
+	KEYIDS["KEY_EPG"]: ("EPG",),
+	KEYIDS["KEY_EXIT"]: ("EXIT",),
+	KEYIDS["KEY_F1"]: ("F1",),
+	KEYIDS["KEY_F2"]: ("F2",),
+	KEYIDS["KEY_F3"]: ("F3",),
+	KEYIDS["KEY_FASTFORWARD"]: ("FASTFORWARD",),
+	KEYIDS["KEY_FAVORITES"]: ("FAVORITES",),
+	KEYIDS["KEY_GREEN"]: ("GREEN",),
+	KEYIDS["KEY_HELP"]: ("HELP",),
+	KEYIDS["KEY_INFO"]: ("INFO",),
+	KEYIDS["KEY_LEFT"]: ("LEFT",),
+	KEYIDS["KEY_MENU"]: ("MENU",),
+	KEYIDS["KEY_MUTE"]: ("MUTE",),
+	KEYIDS["KEY_NEXT"]: ("ARROWRIGHT",),
+	KEYIDS["KEY_OK"]: ("OK",),
+	KEYIDS["KEY_PAUSE"]: ("PAUSE",),
+	KEYIDS["KEY_PLAY"]: ("PLAY",),
+	KEYIDS["KEY_POWER"]: ("POWER",),
+	KEYIDS["KEY_PREVIOUS"]: ("ARROWLEFT",),
+	KEYIDS["KEY_RADIO"]: ("RADIO",),
+	KEYIDS["KEY_RECORD"]: ("RECORD",),
+	KEYIDS["KEY_RED"]: ("RED",),
+	KEYIDS["KEY_REWIND"]: ("REWIND",),
+	KEYIDS["KEY_RIGHT"]: ("RIGHT",),
+	KEYIDS["KEY_STOP"]: ("STOP",),
+	KEYIDS["KEY_TEXT"]: ("TEXT",),
+	KEYIDS["KEY_TV"]: ("TV",),
+	KEYIDS["KEY_UP"]: ("UP",),
+	KEYIDS["KEY_VIDEO"]: ("PVR",),
+	KEYIDS["KEY_VOLUMEDOWN"]: ("VOL-",),
+	KEYIDS["KEY_VOLUMEUP"]: ("VOL+",),
+	KEYIDS["KEY_YELLOW"]: ("YELLOW",)
+}]
 
 def addKeyBinding(domain, key, context, action, flags):
 	keyBindings.setdefault((context, action), []).append((key, domain, flags))
 
-# returns a list of (key, flags) for a specified action
+def removeKeyBinding(key, context, action, wild=True):
+	if wild and action == "*":
+		for ctx, action in keyBindings.keys():
+			if ctx == context:
+				removeKeyBinding(key, context, action, False)
+		return
+	contextAction = (context, action)
+	if contextAction in keyBindings:
+		bind = [x for x in keyBindings[contextAction] if x[0] != key]
+		if bind:
+			keyBindings[contextAction] = bind
+		else:
+			del keyBindings[contextAction]
+
+# Returns a list of (key, flags) for a specified action.
+#
 def queryKeyBinding(context, action):
 	if (context, action) in keyBindings:
 		return [(x[0], x[2]) for x in keyBindings[(context, action)]]
 	else:
-		return [ ]
+		return []
 
 def getKeyDescription(key):
 	if rc_model.rcIsDefault():
 		idx = config.misc.rcused.value
 	else:
-		rctype = config.plugins.remotecontroltype.rctype.value
-		if rctype == 14:	# XP1000
+		# rcType = config.plugins.remotecontroltype.rctype.value
```

Comment:
This will change the config value.

Reply:
Oh dear.  Sorry, I was unaware that the config variables also differed.

Do you want to correct this or should I?

Reply:
Yes .. please let the variable as it is. For now.

Reply:
Okay, the error has been corrected in another commit to the pull request.

Thank you for bringing this error to my attention.  I will make the same correction to the other images.